scripts/scil_connectivity_compute_pca.py

```diff
@@ -150,179 +191,128 @@ def autolabel(rects, axs):
-        nb_sub = [len(dictionary[m]) for m in args.metrics]
-        assert all(x == len(subjects) for x in nb_sub), "Error, the number of matrices for each metric doesn't match" \
-                                                        "the number of subject in the id list." \
-                                                        "Please validate input folder."
+        files_per_metric = [[os.path.join(
+            args.in_folder, '{}_{}.npy'.format(a, m))
+            for a in subjects]
+            for m in args.metrics]
+
+    assert_inputs_exist(parser, sum(files_per_metric, []))
+    matrices_per_metric = [[load_matrix_in_any_format(f) for f in files]
+                           for files in files_per_metric]
 
     # Setting individual matrix shape.
-    mat_shape = dictionary[args.metrics[0]][0].shape
+    mat_shape = matrices_per_metric[0][0].shape
 
-    if args.not_only_common:
-        # Creating input structure using all edges from all subjects.
+    # Creating input structure
+    if args.all_edges:
         logging.info('Creating PCA input structure with all edges...')
-        df = generate_pca_input(dictionary)
     else:
-        m1 = extracting_common_cnx(dictionary, 0)
-        m2 = extracting_common_cnx(dictionary, 1)
-
-        if m1.shape != mat_shape:
-            parser.error("Extracted binary mask doesn't match the shape of individual input matrix.")
-
-        if np.sum(m1) != np.sum(m2):
-            parser.error("Different binary mask have been generated from 2 different metrics, \n "
-                         "please validate input data.")
-        else:
-            logging.info('Data shows {} common connections across the population.'.format(np.sum(m1)))
-
-        dictionary = apply_binary_mask(dictionary, m1)
-
-        # Creating input structure.
         logging.info('Creating PCA input structure with common edges...')
-        df = generate_pca_input(dictionary)
-
-    # Setting 0 values to nan.
-    df[df == 0] = 'nan'
-
-    # Standardized the data.
+        nb_subjects = len(matrices_per_metric[0])
+
+        # Using the first metric
+        subj_masks = [m != 0 for m in matrices_per_metric[0]]  # Mask per subj
+        common_edges_mask = np.sum(subj_masks, axis=0) == nb_subjects
+
+        # Verifying that other metrics have the same common edges
+        for i, matrices in enumerate(matrices_per_metric[1:]):
+            tmp_subj_masks = [m != 0 for m in matrices]  # Binary per subj
+            tmp_common = np.sum(tmp_subj_masks, axis=0) == nb_subjects
+            if not np.array_equal(common_edges_mask, tmp_common):
+                parser.error("Different binary masks (common edge) have been "
+                             "generated from metric {} than other metrics. "
+                             "Please validate your input data."
+                             .format(args.metrics[i]))
+
+        plt.figure()
+        plt.imshow(common_edges_mask)
+        plt.title(common_edges_mask)
```

Comment:
This is returning an error (I guess it's a typo!). I like the addition of the matrix visualization of the common connection by the way, great addition!